include/istio/control/tcp/request_handler.h

```diff
@@ -29,6 +29,9 @@ class RequestHandler {
  public:
   virtual ~RequestHandler() {}
 
+  // Builds shared attributes required for both Check and Report calls
+  virtual void BuildSharedAttributes(CheckData* check_data) = 0;
```

Comment:
I kept `CheckData` as arg here but open to changing the interface to be completely independent of Check call. I was trying to do the minimal changes needed to get the PR out and get feedback.

Reply:
You can call it BuildCheckAttributes since this just moves all the build logic for the initial attributes into a separate function.

Reply:
Good point, will do that!